### PR TITLE
Add support for Adafruit Speaker Bonnet

### PR DESCRIPTION
**Pull request by jasonmj**
I'm submitting this PR as a first step to adding support for the Adafruit Speak Bonnet: 
https://www.adafruit.com/product/3346

It's the first custom system I've made, so this code will definitely need some review before (and if) you decide to merge it into the nerves systems. I personally have used this code on an rpi0, so that's why I'm submitting the PR here. I think it should work fine with the other targets too, but I only have a bbb and rpi3 to test on. If things look good with this PR, I can make some others for the other target systems. 

Another note about using this new overlay, you have to include a custom `config/config.txt` in your project in order to get the Speaker Bonnet to work. Here are the changes needed there: 

```
 # Enable I2C, SPI, and audio
-dtparam=i2c_arm=on
+#dtparam=i2c_arm=on
 dtparam=spi=on
-dtparam=audio=on
+#dtparam=audio=on
```
and 
```
+# Enable MMAP Support
+dtoverlay=i2s-mmap
+
+# Enable HiFi Berry Device Tree Overlay
+dtoverlay=hifiberry-dac
```

The official Adafruit instructions for usage are here: 
https://learn.adafruit.com/adafruit-speaker-bonnet-for-raspberry-pi/raspberry-pi-usage

Note that the Adafruit instructions were adapted from these Pimoroni instructions: 
https://learn.pimoroni.com/tutorial/phat/raspberry-pi-phat-dac-install

In those instructions, you may notice that the script used writes a custom asound config file. I've added this to my project in `rootfs_overlay/etc/asound.conf`: 

```
pcm.speakerbonnet {
   type hw card 0
}

pcm.dmixer {
   type dmix
   ipc_key 1024
   ipc_perm 0666
   slave {
     pcm "speakerbonnet"
     period_time 0
     period_size 1024
     buffer_size 8192
     rate 44100
     channels 2
     format S16_LE
   }
}

ctl.dmixer {
    type hw card 0
}

pcm.softvol {
    type softvol
    slave.pcm "dmixer"
    control.name "PCM"
    control.card 0
}

ctl.softvol {
    type hw card 0
}

pcm.!default {
    type             plug
    slave.pcm       "softvol"
}
```

I'd be happy to chat more about this and help to get it merged. Please feel free to reach out.

### CODE REVIEW

File: fwup.conf [hunk 1]
```diff
@@ -136,6 +136,12 @@ file-resource rootfs.img {
     # Error out if the rootfs size exceeds the partition size
     assert-size-lte = ${ROOTFS_A_PART_COUNT}
 }
+file-resource i2c-bcm2708.dtbo {
```
Comment (fhunleth) [inline fwup.conf:139]: I'm looking at https://github.com/raspberrypi/linux/blob/rpi-5.4.y/arch/arm/boot/dts/overlays/i2c-bcm2708-overlay.dts, and I'm not sure why this is needed. Does the Speaker Bonnet work without it?

**Review by fhunleth — COMMENTED** [review]
Thanks for submitting this. I had a couple questions, but they're mostly so that I understand what's happening better.

**Comment by jasonmj** [comment]
Hi Frank, thanks for your response. I'll try rebuilding without these
elements to see if they are indeed needed. I added them based on my reading
of the Adafruit and Pimoroni instructions (plus a number of other forum
posts that discuss usage of the Hifiberry DAC).

I'll try rebuilding without them and test whether the Speaker Bonnet still
works.

-=-
Jason Johnson
FullSteam Labs
828.202.5300

**Comment by jasonmj** [comment]
OK, I've removed the dependencies in question and tested to make sure the speaker bonnet still works. My latest commit (https://github.com/nerves-project/nerves_system_rpi0/pull/214/commits/168705eef016cef4fe2a62733d45d3ae9a9bc896) contains the changes that remove unused deps.

**Comment by fhunleth** [comment]
Looks good to me. It will be in the next system release.

**Comment by fhunleth** [comment]
Do you have plans to write up a blog post or docs for this? Not that you have to, but if you ever do, I'd want to point people to it.

**Comment by jasonmj** [comment]
I do have plans for a blog post. I've also got an example project I'm planning to clean up and release. :) 

Thanks for your help getting this merged!

**Pull request merged**